WordPress/WordPressTest/PostRepositoryTests.swift

```diff
@@ -312,6 +312,123 @@ class PostRepositoryTests: CoreDataTestCase {
         await fulfillment(of: [cancelled], timeout: 0.3)
     }
 
+    func testFetchTwoFullPages() async throws {
+        // `fetchAllPages` fetchs 100 page instances at at time. Here we simulate two full pages result.
+        remoteMock.remotePostsToReturnOnSyncPostsOfType = try (1...2).map { pageNo in
+            try (1...100).map {
+                let post = try XCTUnwrap(RemotePost(siteID: NSNumber(value: pageNo * 1000 + $0), status: "publish", title: "Post: Test", content: "This is a test post"))
+                post.type = "page"
+                return post
+            }
+        } + [[]]
+
+        let allPages = try await repository.fetchAllPages(statuses: [.publish], in: blogID).value
+        XCTAssertEqual(allPages.count, 200)
+    }
+
+    func testFetchManyManyPages() async throws {
+        // Here we simulate a site that has a super large number of pages.
+        remoteMock.remotePostsToReturnOnSyncPostsOfType = try (1...99).map { pageNo in
+            try (1...100).map {
+                let post = try XCTUnwrap(RemotePost(siteID: NSNumber(value: pageNo * 1000 + $0), status: "publish", title: "Post: Test", content: "This is a test post"))
+                post.type = "page"
+                return post
+            }
+        } + [[]]
+
+        let allPages = try await repository.fetchAllPages(statuses: [.publish], in: blogID).value
+        XCTAssertEqual(allPages.count, 9_900)
+    }
```

Comment:
This test takes a minute to run in CI, according to [Test Analytics](https://buildkite.com/organizations/automattic/analytics/suites/wordpress-ios/runs/cb3293d1-bd68-4119-80f6-6121207598b8):

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/33147f40-a580-4359-a4ec-ffb8d74fca8e)

I like the idea of having it but at this time, unless there's something critical we want to put under test with loading many pages, I wonder whether the time hit is worth the information the test gives us.

Reply:
Oh man... I know it's bad, but I didn't know it's going to be that bad. I think it took like less than 10 seconds to run on my machine. I'll disable this test for now and potentially resurrect it once we've migrated to Apple Silicon agents.

In terms of its purpose, I want to ensure this new predicate `NSPredicate(format: "NOT (SELF IN %@)", allPages.map { $0.objectID })` doesn't break things, when the `allPages` contains thousands of objects.

Reply:
> In terms of its purpose, I want to ensure this new predicate NSPredicate(format: "NOT (SELF IN %@)", allPages.map { $0.objectID }) doesn't break things, when the allPages contains thousands of objects.

Is this because we want to be extra careful and be sure that power users with heaps of pages remain well served, or is there a particular concern in passing a huge list of ids to the predicate? I think both are legit, by the way.

I look at Test Analytics from time to time, and always notice a group of tests that are slow but I can't justify removing, like `[CoreDataMigrationTests testMigrate19to31Success](https://buildkite.com/organizations/automattic/analytics/suites/wordpress-ios/tests/96606287/executions/32750390478)` at 14s or `[ContextManagerTests testIterativeMigration130ToLatest](https://buildkite.com/organizations/automattic/analytics/suites/wordpress-ios/tests/96606254/executions/32750379997)` at 5s. Maybe we could move them to a test suite that runs on a schedule? "Extra careful tests just in case and for code that doesn't change often"

Reply:
I'd say wait for a big longer to see how they perform in the new agents. Maybe they can complete in a tolerable time. 🤞